airbyte-integrations/connectors/source-faker/source_faker/streams.py

```diff
@@ -4,21 +4,37 @@
 
 import datetime
 import os
+from multiprocessing import Pool, current_process
 from typing import Any, Dict, Iterable, Mapping, Optional
 
-from airbyte_cdk.models import AirbyteEstimateTraceMessage, AirbyteTraceMessage, EstimateType, TraceType
+from airbyte_cdk.models import AirbyteRecordMessage, Type
 from airbyte_cdk.sources.streams import IncrementalMixin, Stream
 from mimesis import Datetime, Numeric, Person
 from mimesis.locales import Locale
 
-from .utils import format_airbyte_time, read_json
+from .airbyte_message_with_cached_json import AirbyteMessageWithCachedJSON
+from .utils import format_airbyte_time, generate_estimate, now_millis, read_json
+
+
+class FakerMultithreaded:
+    def worker_init(self):
+        """For the workers, we want a unique instance of the faker packages with their own seeds to differentiate the generated responses"""
+        global person
```

Comment:
why are these global? wouldn't every instance of `FakerMultithreaded` children (e.g: `Users`) have their own instance anyways? or is the goal to have a single static instance of person/datetime/numeric shared across all ?

Reply:
Comment added to code after much experimentation:

>         Note: the instances of the mimesis generators need to be global.
>         Yes, they *should* be able to be instance variables on this class, which should only instantiated once-per-worker, but that's not quite the case:
>         * relying only on prepare as a pool initializer fails because we are calling the parent process's method, not the fork
>         * Calling prepare() as part of generate() (perhaps checking if self.person is set) and then `print(self, current_process()._identity, current_process().pid)` reveals multiple object IDs in the same thread, resetting the internal random counters